guava/src/com/google/common/base/RandomStringUtils.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2011 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.common.base;
+
+import java.util.Random;
+
+/**
+ * Provides API for random String generation
+ * @author Petr Havlicek
+ */
+
+public class RandomStringUtils {
+    
+    private static final char ASCII_PRINT_START = 32;
+    private static final char ASCII_PRINT_END = 126;
+    private static final char ASCII_NUMERIC_START = 48;
+    private static final char ASCII_NUMERIC_END = 57;
+    
+    /**
+     * 
+     * @param len length of the returned String
+     * @param chars characters to be used - if chars contains non-alphabetic non-printable characters, they may appear in the returned String
+     * @return String of length len from characters chars
+     */
+    public static String random(int len, char[] chars) {
+        if (chars == null || chars.length == 0) {
+            throw new IllegalArgumentException("The chars array must not be empty");
+        }
+        Random random = new Random();
+        char[] res = new char[len];
+        for (int i = 0; i < len; i++) {
+            res[i] = chars[random.nextInt(chars.length)];
+        }
+        return new String(res);
+    }
+
+    /**
+     * 
+     * @param len length of the returned String
+     * @param chars String od characters to be used - if chars contains non-alphabetic non-printable characters, they may appear in the returned String
+     * @return String of length len from characters chars
+     */
+    public static String random(int len, String chars) {

```

Comment:
@liach Ok! Agree with this. 
Also charAt() can be more effective than getChars() with short CharSequence.